mirdata/beatport_key.py

```diff
@@ -0,0 +1,348 @@
+        return load_audio(self.audio_path)
+
+    def to_jams(self):
+        """Jams: the track's data in jams format"""
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            metadata={
+                'artists': self.artists,
+                'genres': self.genres,
+                'tempo': self.tempo,
+                'title': self.title,
+                'key': self.key,
+            },
+        )
+
+
+def load_audio(audio_path):
+    """Load a beatport_key audio file.
+    Args:
+        audio_path (str): path to audio file
+    Returns:
+        y (np.ndarray): the mono audio signal
+        sr (float): The sample rate of the audio file
+    """
+    if not os.path.exists(audio_path):
+        raise IOError("audio_path {} does not exist".format(audio_path))
+    return librosa.load(audio_path, sr=None, mono=True)
+
+
+def find_replace(directory, find, replace, pattern):
+    """
+    Replace in some directory all the songs with the format pattern find by replace
+    Parameters
+    ----------
+    directory (str) path to directory
+    find (str) string from replace
+    replace (str) string to replace
+    pattern (str) regex that must match the directories searrched
+    """
+    for path, dirs, files in os.walk(os.path.abspath(directory)):
+        for filename in fnmatch.filter(files, pattern):
+            filepath = os.path.join(path, filename)
+            with open(filepath) as f:
+                s = f.read()
+            s = s.replace(find, replace)
+            with open(filepath, "w") as f:
+                f.write(s)
+
+
+def download(
+    data_home=None, force_overwrite=False, cleanup=True, partial_download=None
+):
+    """Download the beatport_key Dataset (annotations).
+    The audio files are not provided due to copyright issues.
+    Args:
+        data_home (str):
+            Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+        force_overwrite (bool):
+            Whether to overwrite the existing downloaded data
+        cleanup (bool):
+            Whether to delete the zip/tar file after extracting.
+        partial_download(list of str)
+            arguments can be 'audio' 'metadata' or/and 'keys'
+    """
+
+    # use the default location: ~/mir_datasets/beatport_key
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    download_message = ""
+
+    download_utils.downloader(
+        data_home,
+        remotes=REMOTES,
+        partial_download=partial_download,
+        info_message=download_message,
+        force_overwrite=force_overwrite,
+        cleanup=cleanup,
+    )
+    # This dataset annotations have characters that doesnt correspond with json format. In particular, "bpm": nan
+    # doesn't correspond to json format.
+    # input file
+    find_replace(os.path.join(data_home, "meta"), ": nan", ": null", "*.json")
```

Comment:
what is the role of this function when you have an index for all the files in the dataset?

Reply:
Original JSON files have `nan` that are not of the JSON standard format. So I can't read JSON. So from what is downloaded of Zenodo is changed each `nan` to `none`.

Reply:
I understand why the function is there now. Maybe it would be more informative adding the long comment on the readme of the function above and leaving a short one here. Also, thinking out loud, I guess is a good choice to do it in the download so you do it only once and in a consistent way to the index, right?

Reply:
yes, right